test/scalability/mzcompose.py

```diff
@@ -320,10 +347,49 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
     df_workloads = pd.DataFrame(data={"workload": workload_names})
     df_workloads.to_csv(RESULTS_DIR / "workloads.csv")
 
+    endpoint_result_data_by_workload_name: dict[
+        str, dict[Endpoint, WorkloadResult]
+    ] = dict()
+
     for workload in workloads:
         assert issubclass(workload, Workload), f"{workload} is not a Workload"
+        result_data_of_current_workload = dict()
+        endpoint_result_data_by_workload_name[
+            workload.__name__
+        ] = result_data_of_current_workload
+
         for endpoint in endpoints:
-            run_workload(c, args, endpoint, schema, workload())
+            result = run_workload(c, args, endpoint, schema, workload())
+            result_data_of_current_workload[endpoint] = result
+
+    handle_regression_detection(
+        args, regression_baseline_endpoint, endpoint_result_data_by_workload_name
+    )
+
+
+def handle_regression_detection(
+    args: argparse.Namespace,
+    regression_baseline_endpoint: Endpoint | None,
+    endpoint_result_data_by_workload_name: dict[str, dict[Endpoint, WorkloadResult]],
+) -> None:
+    if regression_baseline_endpoint is None:
+        print("No regression detection because '--regression-against' param is not set")
+    else:
+        outcome = create_result_analyzer(args).determine_regression(
+            regression_baseline_endpoint, endpoint_result_data_by_workload_name
+        )
+
+        if outcome.has_regressions():
+            print(
+                f"ERROR: The following regressions were detected (baseline: {regression_baseline_endpoint}):\n{outcome}"
```

Comment:
during benchmarking, `mz_version()` is used to obtain the actual version + git sha being benchmarked. Can we display that here? 

In the jupyter notebook, you can see that `git log` to fetch the description of the commit as  well. You can take the short piece of code and convert it into a library function.

Reply:
✅ 

I also fixed the computation description, which most likely was not working for Postgres.